### PR TITLE
Add a simple `decompileCFR` task

### PR DESCRIPTION
**Pull request by modmuss50**
I have seen a number of people trying to use enigma's export source with mixed results. 

This provides a quick and simple way to export the source.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -667,6 +675,17 @@ task genFakeSource(dependsOn: ["buildYarnTiny", "mapNamedJar"]) {
 	}
 }
 
+task decompileCFR(type: JavaExec, dependsOn: "mapNamedJar") {
+	classpath = configurations.decompileClasspath
+	main = "org.benf.cfr.reader.Main"
+
+	args namedJar.getAbsolutePath(), "--outputdir", file("src").absolutePath
```
Comment (haykam821) [inline build.gradle]: An option should be exposed to change the output path from something other than `./src`.

File: build.gradle [hunk 2]
```diff
@@ -667,6 +675,17 @@ task genFakeSource(dependsOn: ["buildYarnTiny", "mapNamedJar"]) {
...
+
+	doFirst {
+		file("src").deleteDir()
```
Comment (haykam821) [inline build.gradle]: Shouldn't this be `minecraftSrc` or similar to be in line with `remappedSrc`, and in certain scenarios, not overwrite an existing `src` folder?

**Review by haykam821 — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 1 above)
Comment (modmuss50) [reply build.gradle]: I dont think thats needed, adds added compliexity for something that I dont think has a lot of gain.

File: build.gradle (hunk 2 above)
Comment (modmuss50) [reply build.gradle]: there isnt an existing src, I did kinda think of dumping it in src/main/java and making it importable into an ide so you could explore the code a bit easier but thats prob not needed.

I think what we have here is fine? What do others think?

Comment (modmuss50) [reply build.gradle]: on 2nd throughts, remappedSrc does make more sense. I will change this

Comment (haykam821) [reply build.gradle]: Another solution would be to potentially halt the process if `src` already exists.

File: build.gradle (hunk 1 above)
Comment (haykam821) [reply build.gradle]: If there's already an existing `src`, this is what users will have to use without an option:

```bash
mv src tempSrc
./gradlew decompileCFR
mv src minecraftSrc
mv tempSrc src
```

File: build.gradle (hunk 2 above)
Comment (modmuss50) [reply build.gradle]: changed to `namedSrc` to follow the jar.

**Review by hYdos — APPROVED** [review]

**Review by YanisBft — APPROVED** [review]

**Pull request merged**